### PR TITLE
React Native Livestream Application using Agora

### PR DESCRIPTION
**Pull request by zolomohan**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      

#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tool:** [Grammarly](https://grammarly.com)
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by zolomohan** [comment]
This fixes #583

**Comment by hectorkambow** [comment]
Plagiarism check done

**Comment by hectorkambow** [comment]
👋  Thank you for submitting this PR - we will review it ASAP - all PRs go into our queue will be looked over before being approved to be published.

**Comment by zolomohan** [comment]
Great, Waiting for the review. I have one more article almost ready, waiting for this review to complete so that I can apply the suggestions from here on that article too.

**Comment by hectorkambow** [comment]
@zolomohan 
Just wanted to make sure the images did not need any sources - along with the hero image? If you made the hero image that fine. 👍  Otherwise we can grab one from unsplash.com

**Comment by zolomohan** [comment]
The images included are screenshots and the hero image, I grabbed one from Unsplash and added the logos.

File: articles/react-native-agora-livestreaming-app/index.md [hunk 1]
```diff
@@ -0,0 +1,681 @@
...
+title: React Native Livestream Application using Agora
+description: This tutorial will give readers a detailed guide on how they can build an app to create or join a live stream using Agora.
+author: mohan-raj
+date: 2020-11-19T00:00:00-10:00
+topics: []
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/react-native-agora-livestreaming-app/hero.jpg
+    alt: Python example image PyPi
```
Comment (sudosoph) [inline articles/react-native-agora-livestreaming-app/index.md:15]: Alt comment should be related to hero image, i.e. "React Native Agora Livestreaming App"

File: articles/react-native-agora-livestreaming-app/index.md [hunk 2]
```diff
@@ -0,0 +1,681 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/react-native-agora-livestreaming-app/
+title: React Native Livestream Application using Agora
+description: This tutorial will give readers a detailed guide on how they can build an app to create or join a live stream using Agora.
+author: mohan-raj
+date: 2020-11-19T00:00:00-10:00
+topics: []
```
Comment (sudosoph) [inline articles/react-native-agora-livestreaming-app/index.md:10]: Possible Topics: React, Agora, Livestreaming

File: articles/react-native-agora-livestreaming-app/index.md [hunk 3]
```diff
@@ -0,0 +1,681 @@
...
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/react-native-agora-livestreaming-app/hero.jpg
+    alt: Python example image PyPi
+---
+Want to learn how to build a cool and simple Livestreaming app using React Native? In this tutorial, we will do so using [Agora](https://www.agora.io/en/about-us/), which is a video, voice, and live interactive streaming platform. It includes embedded voice and video chat, real-time recording, interactive live streaming, and real-time messaging.
```
Comment (sudosoph) [inline articles/react-native-agora-livestreaming-app/index.md:17]: More context needed on Agora. Who built it? How did it arise? Why did they pick this particular functionality when other libraries exist out there? It helps the reader to know why we pick specific tools, since we are addressing articles to beginners. You can be brief, but some of the info from the page you so usefully point out should be introduced here.

File: articles/react-native-agora-livestreaming-app/index.md [hunk 4]
```diff
@@ -0,0 +1,681 @@
...
+<!--more-->
+Keep reading to find out.
```
Comment (sudosoph) [inline articles/react-native-agora-livestreaming-app/index.md:19]: Unnecessary

File: articles/react-native-agora-livestreaming-app/index.md [hunk 5]
```diff
@@ -0,0 +1,681 @@
...
+
+### Goals
+By the end of this tutorial, you’ll understand
```
Comment (sudosoph) [inline articles/react-native-agora-livestreaming-app/index.md:22]: you'll know:

Don't forget the colon!

File: articles/react-native-agora-livestreaming-app/index.md [hunk 6]
```diff
@@ -0,0 +1,681 @@
...
+
+- How to create/join a live stream using Agora.
```
Comment (sudosoph) [inline articles/react-native-agora-livestreaming-app/index.md:24]: Will put a comment in general rather than inline, but livestream is one word and should consistently be one word throughout. Verify every instance!

File: articles/react-native-agora-livestreaming-app/index.md [hunk 7]
```diff
@@ -0,0 +1,681 @@
...
+
+- How to share a room code for others to join the live stream.
+
+- How to add event listeners on the live stream to listen to various state changes.
+
+### Prerequisites
+This article will not cover tutorial aspects of how React/React Native works. If you don't know how to work with it, please refer to [some tutorials](https://reactnative.dev/docs/tutorial) before beginning with this project.
```
Comment (sudosoph) [inline articles/react-native-agora-livestreaming-app/index.md:31]: Since we are targeting beginner/intermediates, they need to know if they will be able to follow along. So you can rephrase this to be something like: "The fundamentals of React and React Native will not be covered in this tutorial, however, no prior knowledge is required to follow along in building this app" or something similar

File: articles/react-native-agora-livestreaming-app/index.md [hunk 8]
```diff
@@ -0,0 +1,681 @@
...
+
+### Agora
+Agora provides the building blocks for a wide range of real-time engagement possibilities. Agora is a paid service, but don’t worry. The first 10,000 minutes are free every month. You can check their pricing [here](https://www.agora.io/en/pricing/).
```
Comment (sudosoph) [inline articles/react-native-agora-livestreaming-app/index.md:34]: "Agora provides the building blocks for a wide range of real-time engagement possibilities" -> this seems more like marketing-speak. Explain in your own words, or elaborate as suggested above, on the relative benefits of this tool.

"Agora is a paid service, but don’t worry." -> phrasing: keep a formal tone, remove "but don't worry" and instead replace with something like "for the purposes of this tutorial, you can use the free version. The first 10,000 min..."

Comment (sudosoph) [reply articles/react-native-agora-livestreaming-app/index.md:34]: OR even "the free version will suffice" - you get the idea

File: articles/react-native-agora-livestreaming-app/index.md [hunk 9]
```diff
@@ -0,0 +1,681 @@
...
+
+Using Agora, we can develop a variety of applications that need real-time engagement.
+
```
Comment (sudosoph) [inline articles/react-native-agora-livestreaming-app/index.md:37]: What does real-time engagement imply, for users who have never seen the term? Is it always audio/video or is it the general implication of interactivity?

Comment (sudosoph) [reply articles/react-native-agora-livestreaming-app/index.md:37]: that need -> that use, that require

**Comment by zolomohan** [comment]
Hey @sudosoph, thanks for taking the time to review my article. I'll go through the suggestions and resolve them ASAP.

File: articles/react-native-agora-livestreaming-app/index.md (hunk 2 above)
Comment (zolomohan) [reply articles/react-native-agora-livestreaming-app/index.md]: Fixed.

File: articles/react-native-agora-livestreaming-app/index.md (hunk 5 above)
Comment (zolomohan) [reply articles/react-native-agora-livestreaming-app/index.md:22]: Fixed.

File: articles/react-native-agora-livestreaming-app/index.md (hunk 6 above)
Comment (zolomohan) [reply articles/react-native-agora-livestreaming-app/index.md:24]: Fixed.

File: articles/react-native-agora-livestreaming-app/index.md (hunk 4 above)
Comment (zolomohan) [reply articles/react-native-agora-livestreaming-app/index.md:19]: Do you mean just the "Keep reading to find out"?

File: articles/react-native-agora-livestreaming-app/index.md (hunk 7 above)
Comment (zolomohan) [reply articles/react-native-agora-livestreaming-app/index.md:31]: The fundamentals of React and React Native will not be covered in this tutorial. If you don't know the fundamentals, please refer to [some tutorials](https://reactnative.dev/docs/tutorial) before beginning with this project.

This is how I phrased it.

File: articles/react-native-agora-livestreaming-app/index.md (hunk 8 above)
Comment (zolomohan) [reply articles/react-native-agora-livestreaming-app/index.md:34]: "Agora is a paid service, but the first 10,000 minutes are free every month. You can check their pricing [here](https://www.agora.io/en/pricing/)."

Updated it.

They do not have a free version, they have a pay as you go system.

File: articles/react-native-agora-livestreaming-app/index.md (hunk 4 above)
Comment (zolomohan) [reply articles/react-native-agora-livestreaming-app/index.md]: removed "Keep reading to find out".

**Comment by zolomohan** [comment]
Hey @sudosoph, I've updated the article based on your suggestions :)

File: articles/react-native-agora-livestreaming-app/index.md (hunk 2 above)
Comment (hectorkambow) [reply articles/react-native-agora-livestreaming-app/index.md]: 👀  Oh just saw this comment - we typically leave the topics blank (front matter is handled by managers) unless if fits into an existing topic 👍

Comment (zolomohan) [reply articles/react-native-agora-livestreaming-app/index.md]: Oh, okay. I'll remove them.
